### ISSUE TITLE
setting  seed  can't reproduce the results

### ISSUE THREAD

**Issue by BookChan** [NVIDIA-Merlin/HugeCTR#25]
# What have I done?
- set seed in config file:  "solver": {"seed": 100}
- set maxiter=2 ， eval_interval=1
- set file_list.txt with 1 file  set file_list_with.txt with 1 file 
- set train and eval  reader chunk_size with 1 : data_reader.reset(new DataReader<TypeKey>(source_data, batch_size, label_dim, dense_dim,
                                                      check_type, data_reader_sparse_param_array,
                                                      gpu_resource_group, 1, use_mixed_precision));

- run the train process ./huge_ctr --train model.json   twice

# What happened?
AverageLoss(1.200125 and 1.18949 ) are  too far  

the first train log:
```
 [05d17h49m17s][HUGECTR][INFO]: Iter: 1 Time(1 iters): 0.101207s Loss: 1.211278 lr:0.000100

[05d17h49m18s][HUGECTR][INFO]: Evaluation, AUC: 0.501446
[05d17h49m18s][HUGECTR][INFO]: Evaluation, AverageLoss: 1.200125

```
the  second  train log:
```
 
[05d17h51m37s][HUGECTR][INFO]: Iter: 1 Time(1 iters): 0.093456s Loss: 1.200530 lr:0.000100

[05d17h51m37s][HUGECTR][INFO]: Evaluation, AUC: 0.397724

[05d17h51m37s][HUGECTR][INFO]: Evaluation, AverageLoss: 1.18949
```

**Comment by zehuanw** [comment]
Some of the operators in HugeCTR are atomic based. Could you please provide your `model.json`?

**Comment by BookChan** [comment]
model.json:
```
{
  "solver": {
      "lr_policy": "fixed",
      "display": 1,
      "seed": 100,
      "max_iter":  2,
      "gpu": [0],
      "batchsize":  8,
      "snapshot":  1,
      "snapshot_prefix": "./tmp/dcn",
      "eval_interval": 1,
      "batchsize_eval":8,
      "eval_metrics": ["AUC:0.9","AverageLoss"],
      "eval_batches":  1,
      "input_key_type":"I64"
    },
    
    "optimizer": {
      "type": "Adam",
      "global_update": false,
      "adam_hparam": {
        "learning_rate": 0.0001,
        "beta1": 0.9,
        "beta2": 0.999,
        "epsilon": 0.00000001
      }
    },
  
    "layers": [ 
    {
        "name": "data",
        "type": "Data",
        "source": "./tmp/file_list.txt",
        "eval_source": "./tmp/file_list_test.txt",
        "check": "Sum",
        "label": {
          "top": "label",
          "label_dim": 1
        },
        "dense": {
          "top": "dense",
          "dense_dim": 0
        },
        "sparse": [
          {
            "top": "data1",
            "type": "DistributedSlot",
            "max_feature_num_per_sample": 180,
            "slot_num": 50
          }        
        ]
      },
      
      
      {
        "name": "sparse_embedding1",
        "type": "DistributedSlotSparseEmbeddingHash",
        "bottom": "data1",
        "top": "sparse_embedding1",
        "sparse_embedding_hparam": {
          "max_vocabulary_size_per_gpu": 240000,
          "load_factor": 0.75,
          "embedding_vec_size": 16,
          "combiner": 1
        }
      },
  
      {
        "name": "reshape1",
        "type": "Reshape",
        "bottom": "sparse_embedding1",
        "top": "reshape1",
        "leading_dim": 800
      },
      
  
      {
        "name": "slice1",
        "type": "Slice",
        "bottom":  "reshape1"     ,
        "ranges": [[0,800], [0,800]],
        "top": ["slice11", "slice12"]
      },
    
  
  
      {
        "name": "multicross1",
        "type": "MultiCross",
        "bottom": "slice11",
        "top": "multicross1",
        "mc_param": {
          "num_layers": 3
        }
      },
    
      {
        "name": "fc1",
        "type": "InnerProduct",
        "bottom":  "slice12" ,
        "top": "fc1",
         "fc_param": {
          "num_output": 256
        }
      },
      {
        "name": "relu1",
        "type": "ReLU",
        "bottom": "fc1",
        "top": "relu1" 
      },
        
      
      {
        "name": "fc2",
        "type": "InnerProduct",
        "bottom":  "relu1" ,
        "top": "fc2",
         "fc_param": {
          "num_output": 128
        }
      },
      {
        "name": "relu2",
        "type": "ReLU",
        "bottom": "fc2",
        "top": "relu2" 
      },
        
     
    
      {
        "name": "fc3",
        "type": "InnerProduct",
        "bottom":  "relu2" ,
        "top": "fc3",
         "fc_param": {
          "num_output": 64
        }
      },
      {
        "name": "relu3",
        "type": "ReLU",
        "bottom": "fc3",
        "top": "relu3" 
      },
        
       
    
      {
        "name": "concat2",
        "type": "Concat",
        "bottom": ["multicross1","relu3"],
        "top": "concat2"
      },
      
      {
        "name": "fc4",
        "type": "InnerProduct",
        "bottom": "concat2",
        "top": "fc4",
         "fc_param": {
          "num_output": 1
        }
      },
      
      {
        "name": "loss",
        "type": "BinaryCrossEntropyLoss",
        "bottom": ["fc4","label"],
        "top": "loss"
      } 
    ]
  }

```

**Comment by zehuanw** [comment]
Hi @BookChan , thank you for your feedback! As python interface is used in HugeCTR instead of json file, I would like to close this issue. But if you encounter more issues please open a new issue for better tracking.
